hathor/event/websocket/factory.py

```diff
@@ -0,0 +1,99 @@
+from typing import Dict, Set
+from uuid import uuid4
+
+from autobahn.twisted.websocket import WebSocketServerFactory
+from structlog import get_logger
+from twisted.internet.task import LoopingCall
+
+from hathor.conf import HathorSettings
+from hathor.event.websocket.event_streaming import EventStreaming
+from hathor.event.websocket.protocol import HathorEventWebsocketProtocol
+from hathor.util import json_dumpb, json_loadb, reactor
+
+settings = HathorSettings()
+logger = get_logger()
+
+
+class EventWebsocketFactory(WebSocketServerFactory):
+    """ Websocket that will handle events
+    """
+
+    protocol = HathorEventWebsocketProtocol
+
+    def buildProtocol(self, id):
+        return self.protocol(self, id)
+
+    def __init__(self, event_storage):
+        self.connections: Set[HathorEventWebsocketProtocol] = set()
+        self.connections_to_stream_events: Dict[str, EventStreaming] = {}
```

Comment:
Why is the key a random id? Should we allow clients to identify themselves?

Reply:
What do you mean? It's not a random id, and clients don't identify themselves. This is actually the client address. But I'll refactor `connections` to be a `Dict[HathorEventWebsocketProtocol, EventStreaming]` instead, that should make it much simpler.

Reply:
Not sure about the previous code, but now each connection has a `client_peer` property. Maybe that's what @msbrogli meant? Anyways, there are no more keys controlled by the factory.